src/main/java/org/commonjava/service/storage/controller/StorageController.java

```diff
@@ -288,14 +288,27 @@ public FileCopyResult copy(FileCopyRequest request)
                 request.getTargetFilesystem() );
 
         final AtomicInteger copied = new AtomicInteger();
+
+        // Update creation (current time) and expiration for copied file
+        final Date current = new Date();
+        final Date expiration;
+        if ( request.getTimeoutSeconds() > 0 )
+        {
+            expiration = new Date( current.getTime() + TimeUnit.SECONDS.toMillis(request.getTimeoutSeconds()) );
+        }
+        else
+        {
+            expiration = null; // never expire
+        }
+
         request.getPaths().forEach( p -> {
             if ( existing.contains( p ) )
             {
                 skipped.add(p);
             }
             else
             {
-                fileManager.copy(request.getSourceFilesystem(), p, request.getTargetFilesystem(), p);
+                fileManager.copy(request.getSourceFilesystem(), p, request.getTargetFilesystem(), p, current, expiration);
```

Comment:
if the source file is expired, is the physical file will be removed ?

Reply:
No. Only both are expired. Actually when both are expired, the physical file may stay for longer time. We use lazy expiration which means if no one access the file, it won't be removed.

Reply:
Ok, LGTM. thanks.